[wip] chore: add beginnings of docs on project upgrades (#14469)
supabase/supabase: apps/docs/pages/guides/platform/project-upgrades.mdx

import Layout from '~/layouts/DefaultGuideLayout'

export const meta = {
  id: 'project-upgrades',
  title: 'Project Upgrades',
  description: 'Upgrading your Supabase project to the latest platform version.',
}

> Supabase projects use the latest software versions of our platform when they're created. Supabase upgrades all projects hosted on our platform with new software builds, if the builds do not involve breaking changes, and if they can be rolled out with minimal disruption. For upgrades involving breaking changes, or requiring extended downtime, projects can be upgraded on-demand through the Supabase dashboard.

Reviewer: ```suggestion
Supabase projects use the latest software versions of our platform when they're created. Supabase automatically upgrades all projects hosted on our platform if the updates do not involve breaking changes and if they can be rolled out with minimal disruption. For upgrades involving breaking changes or requiring extended downtime, projects can be upgraded on-demand through the Supabase dashboard.
```

## Upgrading a Supabase project

TODO (darora): instructions here

## Planning your upgrade

### Misc considerations

- Replication slots will need to be recreated?
- ??

### Estimating downtime requirements

> Postgres version upgrades, especially when they involve major version upgrades, can require extended maintenance windows. They also tend to grow as a function of the database size. 

Reviewer: ```suggestion
Postgres version upgrades can require extended maintenance windows, especially when they are a major version upgrade. The maintenance window also tends to increase relative to the database size. 
```

> Furthermore, after the upgrade has been completed, you'll also want to ensure that your client applications and services are not running into any issues or errors, which should also be budgeted for as part of the maintenance window.

Reviewer: ```suggestion
After the upgrade has completed, you'll also want to ensure that your client applications and services are not running into any issues or errors. This should also be budgeted as part of the maintenance window.
```
